merge.py

```diff
@@ -0,0 +1,72 @@
+import time
+from collections import deque
+
+
+def mergesort(input_list):
+    """Sort list using mergesort methods.
+    TypeError raised if not list. List must contain only numbers."""
+    if type(input_list) is not list:
+        raise TypeError('Entire list must be numbers')
+    if len(input_list) > 1:
+        center = len(input_list)/2
+        left = deque(mergesort(input_list[:center]))
+        right = deque(mergesort(input_list[center:]))
+    else:
+        return list(input_list)
+
+    output = deque([])
+    popA = left.popleft()
+    popB = right.popleft()
```

Comment:
interestingly, deque supports indexing:

```python
popA = left[0]
```
Not sure if this knowledge helps, but it might remove a bit of the if/else nest you have below.